src/syscheckd/db/src/db.cpp

```diff
@@ -95,6 +114,60 @@ void fim_sync_push_msg(const char* msg)
     }
 }
 
+TXN_HANDLE fim_db_transaction_start(const char* table, result_callback_t row_callback, void *user_data)
+{
+    const std::unique_ptr<cJSON, CJsonDeleter> jsInput
+    {
+        cJSON_Parse(table)
+    };
+
+    callback_data_t cb_data = { .callback = row_callback, .user_data = user_data };
+
+    TXN_HANDLE dbsyncTxnHandle = dbsync_create_txn(FIMDB::getInstance().DBSyncHandle(), jsInput.get(), 0,
+                                                   QUEUE_SIZE, cb_data);
+
+    return dbsyncTxnHandle;
+}
+
+FIMDBErrorCodes fim_db_transaction_sync_row(TXN_HANDLE txn_handler, const fim_entry* entry) {
+
+    nlohmann::json json_insert;
+    auto retVal = FIMDB_OK;
+
+    if (entry->type == FIM_TYPE_FILE)
```

Comment:
Is this validation necessary?

Reply:
Yes, as the entry can be a File or a Registry. This validation will be needed if we want this function to be valid for both registries and files.

Reply:
You are right !